src/Annotation/Traits/ActionTrait.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * @author Szyon Kunowski <szymon.kunowski@gmail.com>
+ */
+
+namespace Opstalent\CrudBundle\Annotation\Traits;
+
+use Doctrine\Common\Annotations\AnnotationException;
+use Opstalent\CrudBundle\Annotation\AbstractAnnotation;
+
+/**
+ * Trait ActionTrait
+ * @package Opstalent\CrudBundle
+ */
+trait ActionTrait
+{
+    /**
+     * @var string[]
+     * @Required
+     * @AnnotatedDescription("The property holds all available actions for class")
+     */
+    public $actions = [];
+
+    /**
+     * @var array
+     */
+    protected $availableActions = ['addable', 'getable', 'listable', 'editable'];
+
+    /**
+     * @param string $key
+     * @param array $data
+     * @return array
+     */
+    protected function setActions(string $key, array $data): array
+    {
+        if (array_key_exists($key, $data)) {
+            foreach ($data[$key] as $action) {
+                $this->enableAction($action);
+            }
+        }
+        return $this->getActions();
+    }
+
+    /**
+     * @return array
+     */
+    public function getActions(): array
+    {
+        return $this->actions;
+    }
+
+    /**
+     * @return array
+     */
+    protected function getAvailableActions(): array
+    {
+        return $this->availableActions;
+    }
+
+    /**
+     * @param string $name
+     * @return int
+     */
+    protected function addAvailableAction(string $name): int
+    {
+        return array_push($this->availableActions, $name);
+    }
+
+    /**
+     * @param string $action
+     * @return bool
+     * @throws AnnotationException
```

Comment:
it's inconsistency. method is named `isSomething` so it should return `true` if it is and `false` if it is not.

you are returning `true` if it is and throwing an exception if it is not. where's the pattern?